### PR TITLE
fix(PlanarControls): fix pan movement

### PR DESCRIPTION
**Pull request by mgermerie**
## Description
Fix pan movement in a `PlanarView` context.

### CODE REVIEW

**Comment by gchoqueux** [comment]
there's a [failed test](https://github.com/iTowns/itowns/pull/1571/checks#step:8:493)

File: src/Controls/PlanarControls.js [hunk 1]
```diff
@@ -396,30 +400,22 @@ class PlanarControls extends THREE.EventDispatcher {
...
+        this.camera.position.x = newPosition.x;
+        this.camera.position.y = newPosition.y;
 
-        // lateral movement (local x axis)
-        vect.set(panSpeed * -1 * deltaMousePosition.x, 0, 0);
-        this.camera.position.copy(this.camera.localToWorld(vect));
-
-        // vertical movement (world z axis)
-        const newAltitude = this.camera.position.z + panSpeed * deltaMousePosition.y;
+        // compute ground level under the camera
```
Comment (mgermerie) [inline src/Controls/PlanarControls.js]: This part (checking if destination position does not collide with layer) might be redundant with the check which is done in `update` method. However, if we just test it in `update` method, the camera is able to go under a layer, then move back above it when `update` is called (which is at the next image refreshment).

Comment (gchoqueux) [reply src/Controls/PlanarControls.js]: Could you try to remove altitude checking in `handlePanMovement` and move handling Collision to the `udapte` end?

Comment (mgermerie) [reply src/Controls/PlanarControls.js]: Indeed, moving collision handling to the end of `update` method works as expected.

File: src/Controls/PlanarControls.js [hunk 2]
```diff
@@ -396,31 +396,9 @@ class PlanarControls extends THREE.EventDispatcher {
...
-        // vertical movement (world z axis)
-        const newAltitude = this.camera.position.z + panSpeed * deltaMousePosition.y;
-
-        // check if altitude is valid
-        if (this.groundLevel < newAltitude && newAltitude < this.maxAltitude) {
-            this.camera.position.z = newAltitude;
-        }
+        vect.set(-deltaMousePosition.x, deltaMousePosition.y, 0);
+        const newPosition = new THREE.Vector3().copy(this.camera.localToWorld(vect));
+        this.camera.position.copy(newPosition);
```
Comment (gchoqueux) [inline src/Controls/PlanarControls.js]: could you try?
```js
vect.set(-deltaMousePosition.x, deltaMousePosition.y, 0);
this.camera.localToWorld(vect);
this.camera.position.copy(vect);
```

**Review by gchoqueux — APPROVED** [review]

**Pull request merged**